application/src/main/java/org/thingsboard/server/actors/calculatedField/CalculatedFieldManagerMessageProcessor.java

```
@@ -180,16 +209,35 @@ public void onEntityLifecycleMsg(CalculatedFieldEntityLifecycleMsg msg) throws C
                 }
                 break;
             }
+            case DEVICE_PROFILE:
+            case ASSET_PROFILE: {
+                switch (event) {
+                    case DELETED:
+                        onProfileDeleted(msg.getData(), msg.getCallback());
+                        break;
+                    default:
+                        msg.getCallback().onSuccess();
+                        break;
+                }
+                break;
+            }
             default: {
                 msg.getCallback().onSuccess();
             }
         }
     }
 
+    private void onProfileDeleted(ComponentLifecycleMsg msg, TbCallback callback) {
+        entityProfileCache.removeProfileId(msg.getEntityId());
+        callback.onSuccess();
+    }
+
     private void onEntityCreated(ComponentLifecycleMsg msg, TbCallback callback) {
         EntityId entityId = msg.getEntityId();
         EntityId profileId = getProfileId(tenantId, entityId);
-        cfEntityCache.add(tenantId, profileId, entityId);
+        if (profileId != null) {
```

getProfileId should throw an exception if entity type is invalid